tls/s2n_server_extensions.c

```diff
@@ -49,22 +49,21 @@
 #define s2n_server_can_send_nst(conn) (s2n_server_sending_nst((conn)) && \
         (conn)->actual_protocol_version < S2N_TLS13)
 
-int s2n_server_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+/* compute size server extensions send requires */
+int s2n_server_extensions_send_size(struct s2n_connection *conn)
```

Comment:
`send_size` sounds like it is sending the size of something (which isn't even provided) over the connection. Maybe cather call it `compute_size`.

Btw: In our other project Cameron had a nice solution for the problem, which avoids duplicating too much code and which avoids size checks and actual data writing from diverging. Instead of having a single compute and send method, he introduced a common method which took a writer interface as an argument. For computing the size, we use a writer which just writes the sizes. For sending data, we use a writer which writes the data actually into the buffer.

A bit more ugly to do in C however. You can either provide a callback function as a writer interface, but that degrades performance. Or templatize it via a macro, which is rather ugly.

Reply:
yes we do have the same goals - see step 4 of https://github.com/awslabs/s2n/issues/1641 where the plan is to refactor all these extensions with a unified struct. the intermediate steps is to have correct behavior and good test coverage so we can confidently refactor this.

as for `send_size`, is it a name that has been followed for some of the extensions, so it's probably easier to keep the convention before refactoring them at the later stage.

Reply:
i just realized i didn't address the need to compute the size - yes, that's an option here by first writing into a stuffer, then writing the size of the stuffer. it's nice to have, but may lead to more allocations... trade offs..